src/PrMerger.js

```diff
@@ -35,17 +57,37 @@ class PrMerger {
             try {
                 const rawPr = this._todo.shift();
 
+                scanResult.awakePrs.push(rawPr);
+
                 if (rawPr.labels.some(el => el.name === Config.ignoredByMergeBotsLabel())) {
                     this._ignored++;
                     Logger.info(`Ignoring PR${rawPr.number} due to ${Config.ignoredByMergeBotsLabel()} label`);
                     continue;
                 }
 
+                const clearedForMerge = rawPr.labels.some(el => el.name === Config.clearedForMergeLabel());
+
+                // There are usually few 'cleared' PRs (i.e., ready for merge), so do not ignore them.
+                // This should help handle sutiations where some of such 'cleared' PRs were updated just after
+                // this scan has been started. Other (non-cleared) PRs (that are not going to be merged now anyway)
+                // can be ignored until the next scan.
+                // TODO: also handle a situation when the PR becomes 'cleared' just after this scan began.
+                if (!clearedForMerge && _LastScan && _LastScan.isStillUnchanged(rawPr)) {
```

Comment:
> > would not the updated_at field of the just-cleared PR already reflect the update?!
> 
> What if the PR is updated after `GH.getOpenPrs()` but before the loop iteration on this PR?

... then we would not know that the PR was cleared -- getOpenPrs() would not contain the clearance label.

From GitHub and relevant Anubis code point of view, the clearance label is similar to any other label (and labeling itself is similar to other PR updates that keep a PR open!). I would like us to be able to rely on bare isStillUnchanged() check because if we cannot, then this code will not work correctly in cases other than cleared for merged PRs.

AFAICT, this clearedForMerge check will _speed up_ cleared PR processing in the following cases, but they ought to be very rare (or, if GitHub properly implements atomic PR metadata updates, nonexistent!). The red flags raised by this rare-case optimization are just not worth it IMO.

1. GitHub changes PR label.
2. GitHub sends half-updated PR metadata to Anubis (in response to a getOpenPrs() request).
3. GitHub changes PR updated_at (as the result of the label change in item 1).
4. GitHub sends a notification about the fully-updated PR.

Reply:
Removed the 'clearedForMerge' condition (bf9a882).